designs/build_debug_artifacts.md

```diff
@@ -0,0 +1,120 @@
+SAM Build producing debuggable artifacts
+========================================
+
+What is the problem?
+--------------------
+
+`sam build` produces artifacts that are built for production use. In some langauges (usually interpreted), 
+these production artifacts are also debuggable locally. In the case of compiled languages, you usually need to compile
+the binary or artifact in a specific manner for them to be debuggable.  
+
+What will be changed?
+---------------------
+
+We will introduce a way in `sam build` to produce these debuggable artifacts for those compiled languages.
+
+Success criteria for the change
+-------------------------------
+
+1. Artifacts generated will be debuggable for DotNetCore2.1 Runtimes
+
+Out-of-Scope
+------------
+
+1. Other languages `sam build` supports will not be changed
```

Comment:
its a no-op for other languages im assuming.

Reply:
Yes, for now. We could produce metadata (debug specific) from #988 only under this case but that would be in the future, if we do follow that approach. For now, it will produce the same artifacts as build does today. We could add a log but the artifacts are already debuggable so I don't see a need to fail or do any special handling.